eyeshade/workers/reports.js

```diff
@@ -74,5 +77,14 @@ exports.initialize = async (debug, runtime) => {
 
   if ((typeof process.env.DYNO === 'undefined') || (process.env.DYNO === 'worker.1')) {
     setTimeout(() => { daily(debug, runtime) }, 5 * 1000)
+    updateBalancesOnInterval(runtime)
   }
 }
+
+async function updateBalancesOnInterval (runtime) {
+  await updateBalances(runtime)
+  const now = +(new Date())
+  const hours6 = 1000 * 60 * 60 * 6
+  const msUntilNext = now % hours6
+  setTimeout(() => updateBalancesOnInterval(runtime), msUntilNext)
+}
```

Comment:
I'm not sure this does what you want here.  I think what you're after is scheduling things when the hour % 6 == 0, so the 0th hour of the day, 6th, 12th and so on.

But I don't think we want to set the timeout to the remainder.  Suppose it is 6:01AM.  The remainder would be 1 minute, and it'd get scheduled to run in 1 minute.  That might finish and it's 6:15, at which point we schedule the next job to run in 15 minutes.

Does that make sense or perhaps I'm missing something?

Reply:
ah, i am missing a line aren't i. yeah, i'll update